pkg/block/azure/adapter.go

```diff
@@ -29,14 +29,16 @@ const (
 	// more the 5000 different accounts, which is highly unlikely
 	udcCacheSize = 5000
 
-	BlobEndpointFormat = "https://%s.blob.core.windows.net/"
+	BlobEndpointGeneralFormat    = "https://%s.blob.core.windows.net/"
+	BlobEndpointChinaCloudFormat = "https://%s.blob.core.chinacloudapi.cn/"
 )
 
 type Adapter struct {
 	clientCache        *ClientCache
 	preSignedExpiry    time.Duration
 	disablePreSigned   bool
 	disablePreSignedUI bool
+	chinaCloud         bool
```

Comment:
My suggestion is too not use a boolean but rather allow providing a custom domain with the default being the global Azure blob domain. In addition we might want to log a warning when using a custom domain